src/Entity/CitizenProjectCategory.php

```diff
@@ -21,4 +24,52 @@
  */
 class CitizenProjectCategory extends BaseEventCategory
 {
+    /**
+     * @var CitizenProjectCategorySkill[]|Collection
+     *
+     * @ORM\OneToMany(targetEntity="CitizenProjectCategorySkill", mappedBy="category", indexBy="id", cascade={"all"}, orphanRemoval=true)
+     *
+     * @Assert\Valid
+     */
+    private $categorySkills;
+
+    public function __construct(? string $name = null, ? string $status = parent::DISABLED)
+    {
+        parent::__construct($name, $status);
+
+        $this->categorySkills = new ArrayCollection();
+    }
+
+    public function addSkill(CitizenProjectSkill $skill, bool $promotion = false): void
+    {
+        foreach ($this->categorySkills as $categorySkill) {
+            if ($categorySkill->getSkill() === $skill) {
+                $categorySkill->setPromotion($promotion);
+
+                return;
+            }
+        }
+
+        $categorySkill = new CitizenProjectCategorySkill($this, $skill, $promotion);
+        $this->categorySkills->add($categorySkill);
+    }
+
+    public function addCategorySkill(CitizenProjectCategorySkill $skill): void
+    {
+        if (!$this->categorySkills->contains($skill)) {
+            $this->categorySkills->add($skill);
```

Comment:
here you should add `$skill->setCategory($this);` and remove the event listener in `CitizenProjectCategoryAdmin::configureFormFields`